### DISCUSSION TITLE
「字体平滑」选项是否无法覆盖字体默认的微调（hinting）？

### DISCUSSION THREAD

**Discussion by Wenti-D** [F9y4ng/GreasyFork-Scripts#389] [问答专区(Question / Answer)]
## 问题
我尝试了以微软雅黑字体启用或不启用「字体平滑」选项渲染，但是得到的结果完全一致，并且仍然遵循微软雅黑自己的微调（hinting）。那么「字体平滑」实际上会如何生效？

## 测试图
![屏幕截图 2024-11-03 160813](https://github.com/user-attachments/assets/256d5021-6a07-4575-a513-8c814f7b7809)

![屏幕截图 2024-11-03 160836](https://github.com/user-attachments/assets/c9c8e0c7-ec39-4c96-86b5-b5a40e7e10fd)

## 附加消息

系统是 Windows 11 23H2 64 位；
浏览器是 Chrome 130.0.6723.92 64 位；
微软雅黑较小字号的微调极其难看。

**Comment by F9y4ng** [comment]
字体平滑相对应的 CSS 样式在 windows 下 Chromium 系浏览器的效果只能说是差强人意。在英文字体中，对 M$ 的 `Calibri, Candara` 等字体有优化，但对中文几乎没有任何效果。这在 MDN 中有相关较详细的介绍：`text-rendering`。

其他相关的 CSS 样式代码，也仅在 macOS 下，Safari 和 Firefox 中才能工作。

而更高级别的 CSS 样式，如 2024.11.02.1 版本新增的`font-feature-settings`, `font-variant`属性，会对 OpenType 字体（包含中文）有效，但需要额外设置参数。

**Reply by Wenti-D** [reply]
感谢回复！然而 Opentype 特性也无法控制字体微调（悲）。

**Reply by F9y4ng** [reply]
如果你使用的是高分屏高分辨率可以开启浏览器缩放或脚本的字体缩放来解决windows下字体hinting的问题。

如果是低分屏，可以考虑开启字体缩放，并附加字体阴影，将阴影颜色透明度降低，可以间接实现灰度渲染和字体平滑的效果。当然，通过CSS样式模拟和系统级的调教是无法比拟的。 #160 或许会有些帮助。

**Reply by Wenti-D** [reply]
那有没有可能，像 NVidia 的 DSR 一样，以放大的情形渲染，然后将渲染结果缩回原来的大小呢？

**Reply by F9y4ng** [reply]
几乎没有可能，在浏览器沙盒内的权限非常的有限，唯一可做的就是通过css来模拟灰度渲染，尽可能的减少字体的边缘锐化。但模拟终究是模拟，最终效果还是因人而异。

最根本的解决方案就是高分屏4K分辨率+200％的系统缩放才能完美的解决windows字体渲染的问题。

**Reply by Wenti-D** [reply]
其实我试过 4K 屏幕 + 200% 缩放，还是差点，最终仍然靠 MacType 了事。然而自 Chrome 118 以来 MacType 管不到了，寻求了许多方案都无果。

**Reply by F9y4ng** [reply]
你这么高的要求为什么不用Mac？也只有Mac的渲染质量能满足你了，放弃windows是最佳选择。

**Reply by Wenti-D** [reply]
雀食。
